src/main/java/com/smartbear/postman/collection/PostmanCollectionFactory.java

```diff
@@ -1,16 +1,50 @@
 package com.smartbear.postman.collection;
 
+import com.eviware.soapui.support.UISupport;
 import net.sf.json.JSONObject;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class PostmanCollectionFactory {
     public static final String INFO = "info";
+    public static final String COLLECTION = "collection";
+    public static final String VERSION_2 = "2.0.0";
+    public static final String VERSION_2_1 = "2.1.0";
+    private static final String V1_IS_DEPRECATED_ERROR_MESSAGE =
+            "Postman Collections in v1 format are no longer supported. " +
+            "Convert your collection to version v2 or higher and try importing again.";
+    private static final Logger logger = LoggerFactory.getLogger(PostmanCollectionFactory.class);
 
     public static PostmanCollection getCollection(JSONObject collectionObject) {
-        Object value = collectionObject.get(INFO);
-        if (value == null) {
-            return new PostmanCollectionV1(collectionObject);
+        Object collectionSharedViaAPI = collectionObject.get(COLLECTION);
+        if (collectionSharedViaAPI != null) {
+            collectionObject = (JSONObject) collectionObject.get(COLLECTION);
+        }
+
+        Object infoSection = collectionObject.get(INFO);
+        if (infoSection != null) {
+            return getCollectionByVersion(collectionObject, (JSONObject) infoSection);
         } else {
-            return new PostmanCollectionV2(collectionObject);
+            // info section is not present for collections of version 1.0.0
+            UISupport.showErrorMessage(V1_IS_DEPRECATED_ERROR_MESSAGE);
+            return null;
```

Comment:
I have 2 concerns with this code:
1. You return `null` so we are potentially vulnerable to NPE outside of this code - how have you handled it?
2. I am not sure if using `UISupport` here is a good idea. The plugin should be UI-independent and error flow should be handled differently, for example with Exception. The caller of this code is then responsible to handle the potential Exception and may decide whether to use dialog, logger or some failover.